rest/config_legacy.go

```diff
@@ -236,6 +251,62 @@ func (lc *LegacyServerConfig) ToStartupConfig() (*StartupConfig, DbConfigMap, er
 	return &sc, dbs, nil
 }
 
+func gocbv1tov2AddressUpgrade(server string) (newServer, username, password string, err error) {
+	connSpec, err := connstr.Parse(server)
+	if err != nil {
+		return "", "", "", err
+	}
+
+	splitServers := strings.Split(server, ",")
+	// Only first host is supported to have username and pass or else gocbconnstr.Parse does not work correctly
+	if strings.Contains(splitServers[0], "@") {
+		u, err := url.Parse(splitServers[0])
+		if err != nil {
+			return "", "", "", err
+		}
+		urlUsername := u.User.Username()
+		urlPassword, urlPasswordSet := u.User.Password()
+		if urlUsername != "" && urlPasswordSet {
+			username = urlUsername
+			password = urlPassword
+		}
+	}
+
+	var hosts string
+	for _, addr := range connSpec.Addresses {
+		if connSpec.Scheme == "http" {
+			if addr.Port != 8091 {
+				return "", "", "", fmt.Errorf("automatic connection string conversion does not support non-default host ports. " +
```

Comment:
Users probably won't know what triggered 'automatic connection string conversion' - maybe rephrase as "automatic migration connection string from http:// to couchbase:// scheme doesn't support non-default ports"